### ISSUE TITLE
PlusCal translation incorrectly depends on definition

### ISSUE THREAD

**Issue by dricketts** [tlaplus/tlaplus#338]
In the following PlusCal algorithm, I would expect step1 and step2 to get translated to equivalent formulas:

```
--algorithm Test
{
    variables x = 0, y = 0;

    define {
        x_alias == x
    }

    {
        step1: x := 1;
               y := x;

        step2: x := 1;
               y := x_alias;
    }
}
```

However, step1 translates to:

```
step1 == /\ pc = "step1"
         /\ x' = 1
         /\ y' = x'
         /\ pc' = "step2"
```

while step2 translates to:

```
step2 == /\ pc = "step2"
         /\ x' = 1
         /\ y' = x_alias
         /\ pc' = "Done"
```

Notice that there is no prime on `x_alias`. I'm not sure which translation is correct (I expected `x_alias'` for step2), but it does seem like the two steps should translate to equivalent formulas.

**Comment by lemmy** [comment]
https://github.com/tlaplus/tlaplus/blob/659b1aa4ac4b8504910415c67cdd17b18ed3805f/tlatools/src/pcal/PcalTLAGen.java#L3285-L3296

**Comment by dricketts** [comment]
Fair enough, I'll close the issue. The behavior surprised me, but I should have read the manual more closely. I also found this forum post discussing the same behavior: https://groups.google.com/forum/#!topic/tlaplus/vNPlerNO5yQ.

**Comment by Calvin-L** [comment]
I don't agree that this is expected behavior. The manual _also_ says in the introduction:

> you can omit the labels and let the PlusCal translator add them as necessary. You are likely to do this only for uniprocess (sequential) algorithms, **where the partitioning of the computation into steps does not affect the values computed**

(emphasis mine)

Dan's example is a case where the placement of labels affects the value computed by the sequential algorithm, since

        step2: x := 1;
               y := x_alias;

computes something different from

        step2: x := 1;
        step3: y := x_alias;

**Comment by dricketts** [comment]
I agree with Calvin. I guess I prematurely closed this issue. Re-opening for discussion.

**Comment by lemmy** [comment]
I suggest to continue this discussion at the Google group.

**Comment by dricketts** [comment]
Discussion continued here: https://groups.google.com/forum/#!topic/tlaplus/vNPlerNO5yQ
